### PR TITLE
Update json-and-xml-functors.md to reflect Documentation note for quotes in json functor

### PR DESCRIPTION
**Pull request by archit-harness**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

Original ticket - https://harness.atlassian.net/browse/PIE-7455

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [*] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [*] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md [hunk 1]
```diff
@@ -201,7 +201,18 @@ If we render it using `<+json.format(<+pipeline.stages.Functors.spec.execution.s
...
+```
+NOTE:
+
+1. JSON accepts the control sequence `\n` in strings. Use a prettifier like jq to format the JSON.
+
+2. If you are passing the condition "<+json.object(httpResponseBody).fields.parent>" in double quotes which is considered to be a string.
+   So, "<+json.select("fields.status.name", httpResponseBody)>"=="In Progress" is treated as string comparison and will not work.
+   Rather use -> <+json.select("fields.status.name", httpResponseBody)>=="In Progress"
+   Same goes for null comparison.
+   Correct way is -> <+json.object(httpResponseBody).fields.parent>!=null
```
Comment (SudarshanPurohit) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md:214]: Tweaking it a bit for readability: 

Any expression passed in double quotes is considered a string and is not evaluated. For example, `"<+json.object(httpResponseBody).fields.parent>"` will be just a string, So, a comparison like `"<+json.select("fields.status.name", httpResponseBody)>"=="In Progress"` is treated as string comparison and will not work as expected. 
Instead, use `<+json.select("fields.status.name", httpResponseBody)>=="In Progress"`. 

The keyword `null`, too, shouldn't be enclosed in quotes during comparison. Here's an example of a null comparison: `<+json.object(httpResponseBody).fields.parent>!=null`

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md [hunk 2]
```diff
@@ -201,7 +201,18 @@ If we render it using `<+json.format(<+pipeline.stages.Functors.spec.execution.s
 ```json
 {\n  "data": {\n    "attributes": {\n      "name": "new-construction-api",\n      "version_pins": {\n        "mvn-service://new-construction-api": "0.0.253-feature_NC-6595.8d268cd~nc1.6312a66"\n      }\n    }\n  }\n}\n
 ```
-Note: JSON accepts the control sequence `\n` in strings. Use a prettifier like jq to format the JSON.
+
+```
+NOTE:
+
+1. JSON accepts the control sequence `\n` in strings. Use a prettifier like jq to format the JSON.
```
Comment (rashminandasahoo) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md:208]:
```suggestion
JSON accepts the control sequence `\n` as strings. To format JSON, use `jq` to prettify the JSON.
```

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md [hunk 3]
```diff
@@ -201,7 +201,18 @@ If we render it using `<+json.format(<+pipeline.stages.Functors.spec.execution.s
...
+
+2. If you are passing the condition "<+json.object(httpResponseBody).fields.parent>" in double quotes which is considered to be a string.
```
Comment (rashminandasahoo) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md:210]:
```suggestion
2. Conditional expressions within double quotes are considered strings.
```

Comment (rashminandasahoo) [reply docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md:210]: And then add the examples suggested by @SudarshanPurohit .

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md [hunk 4]
```diff
@@ -201,7 +201,18 @@ If we render it using `<+json.format(<+pipeline.stages.Functors.spec.execution.s
...
+   So, "<+json.select("fields.status.name", httpResponseBody)>"=="In Progress" is treated as string comparison and will not work.
+   Rather use -> <+json.select("fields.status.name", httpResponseBody)>=="In Progress"
```
Comment (rashminandasahoo) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md:212]:
```suggestion
   Use `<+json.select("fields.status.name", httpResponseBody)>=="In Progress"` instead.
```

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md [hunk 5]
```diff
@@ -201,7 +201,18 @@ If we render it using `<+json.format(<+pipeline.stages.Functors.spec.execution.s
...
+   Same goes for null comparison.
```
Comment (rashminandasahoo) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md:213]:
```suggestion
   Similarly, use `<+json.object(httpResponseBody).fields.parent>!=null` for null comparison.
```

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md (hunk 1 above)
Comment (archit-harness) [reply docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md]: done

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md (hunk 2 above)
Comment (archit-harness) [reply docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md]: done

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md (hunk 4 above)
Comment (archit-harness) [reply docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md]: done

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md (hunk 5 above)
Comment (archit-harness) [reply docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md]: done

File: docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md [hunk 6]
```diff
@@ -201,7 +201,17 @@ If we render it using `<+json.format(<+pipeline.stages.Functors.spec.execution.s
...
+
+```
+NOTE:
+
+1. JSON accepts the control sequence `\n` as strings. To format JSON, use `jq` to prettify the JSON.
+
+2. Conditional expressions within double quotes are considered strings.
+   So, "<+json.select("fields.status.name", httpResponseBody)>"=="In Progress" is treated as string comparison and will not work.
+   Use `<+json.select("fields.status.name", httpResponseBody)>=="In Progress"` instead.
+   Similarly, use `<+json.object(httpResponseBody).fields.parent>!=null` for null comparison.
```
Comment (SudarshanPurohit) [inline docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md:213]: The keyword null, too, shouldn't be enclosed in quotes during comparison. Here's an example of a null comparison: <+json.object(httpResponseBody).fields.parent>!=null

Comment (archit-harness) [reply docs/continuous-delivery/cd-technical-reference/cd-gen-ref-category/json-and-xml-functors.md]: done

**Review by SudarshanPurohit — APPROVED** [review]

**Review by rashminandasahoo — APPROVED** [review]

**Pull request merged**